automated-testing/performance-testing/readme.md

```diff
@@ -1,67 +1,357 @@
+identify spikes in disk I/O.
+
+## Processor
+
+### % Processor time
+
+This is the percentage of total elapsed time that the processor was busy
+executing. This counter can either be too high or too low. If your processor
+time is consistently below 40%, then there is a question as to whether you
+have over provisioned your CPU. 70% is generally considered a good target
+number and if you start going higher than 70%, you may want to explore why
+there is high CPU pressure.
+
+### % Privileged time
+
+This measures the percentage of elapsed time the processor spent executing in
+kernel mode.  Since this counter takes into account only kernel operations a
+high percentage of privileged time (greater than 25%) may indicate driver or
+hardware issue that should be investigated.
+
+### % User time
+
+The percentage of elapsed time the processor spent executing in user mode (your
+application code). A good guideline is to be consistently below 65% as you want
+to have some buffer for both the kernel operations mentioned above as well as
+any other bursts of CPU required by other applications.
+
+### Queue Length
+
+This is the number of threads that are ready to execute but waiting for a core
+to become available.  On single core machines a sustained value greater than 2-3
+can mean that you have some CPU pressure.  Similarly, for a multicore machine
+divide the queue length by the number of cores and if that is continuously
+greater than 2-3 there might be CPU pressure.
+
+## Network Adapter
+
+Network speed is often a hidden culprit of poor performance. Finding the root
+cause to poor network performance is often difficult. The source of issues can
+originate from bandwidth hogs such as videoconferencing, transaction data,
+network backups, recreational videos.
+
+Packet loss is another area that contributes to poor performance. It occurs when
+packets fail to reach their destination, either by errors in the data
+transmission or network congestion. Packet loss can also be caused by Dos/DDoS
+attacks.
+
+Some of the tools that can help include:
+
+- ifconfig
+- netstat
+- iperf
+- tcpretrans
+- tcpdump
+- WireShark
+
+For most of these counters below, there are no hard and fast numbers. Wi-Fi
+speeds are lower as compared to wired, ethernet speeds. Networking speeds have
+increased dramatically over the years. We can expect all the numbers below to
+change over time.
+
+| Network Type | Speed  |
+|:-------------------- |:--------------------  |
+| Basic Wifi Speed | 3 to 8 Mpbs  |
+| Moderate Wifi Speed | 10 to 25 Mpbs  |
+| Fast Wifi Speed | 300 to 1000 Mbps  |
+| Fast Ethernet | 100 Mbps  |
+| Gigabit Ethernet | 1,000 Mbps  |
+| 10 Gigabit Ethernet | 10,000 Mbps  |
+
+### Network Counters
+
+The table above gives you some reference points to better understand what you
+can expect out of your network. Here are some counters that can help you
+understand where the bottlenecks might exist:
```

Comment:
I think the metrics below can be explained without the table above

Reply:
Agreed. It made sense to remove it. Put in some basic troubleshooting steps instead.